Added new quickstart docs. (#255)
codeigniter4/shield: docs/2 - quickstart.md

```php
public function loginRedirect(): string
{
    $url = auth()->user()->inGroup('admin')
        ? '/admin'
        : setting('Auth.redirects')['login'];

    return $this->getUrl($url);
}
```

### Customize register redirect

You can customize where a user is redirected to after registration in the `registerRedirect` method of the `Auth` config file.

```php
public function registerRedirect(): string
{
    $url = setting('Auth.redirects')['register'];

    return $this->getUrl($url);
}
```

### Customize logout redirect

The logout redirect can also be overridden by the `logoutRedirect` method of the `Auth` config file. This will not be used as often as login and register, but you might find the need. For example, if you programatically logged a user out you might want to take them to a page that specifies why they were logged out. Otherwise, you might take them to the home page or even the login page.

```php
public function logoutRedirect(): string
{
    $url = setting('Auth.redirects')['logout'];

    return $this->getUrl($url);
}
```

### Customize Remember-me functionality

> Remember-me functionality is enabled by default for the `Session` handler. While this is handled in a secure manner, some sites may want it disbled. You might also want to change how long it remberers a user and doesn't require additional login.

Author: You typoed my typo :) 

```suggestion
Remember-me functionality is enabled by default for the `Session` handler. While this is handled in a secure manner, some sites may want it disabled. You might also want to change how long it remembers a user and doesn't require additional login.
```

```php
public array $sessionConfig = [
    'field'              => 'user',
    'allowRemembering'   => true,
    'rememberCookieName' => 'remember',
    'rememberLength'     => 30 * DAY,
];
```

### Change Access Token Lifetime

By default, Access Tokens can be used for 1 year since they last use. This can be easily modified in the `Auth` config file.

```php
public int $unusedTokenLifetime = YEAR;
```

### Enable Two-Factor Authentication

Turned off by default, Shield's Email-based 2FA can be enabled by specifying the class to use in the `Auth` config file.

```php
public array $actions = [
    'login'    => \CodeIgniter\Shield\Authentication\Actions\Email2FA,
    'register' => null,
];
```

### Enable Account Activation via Email

By deafult, once a user registers they have an active account that can be used. You enable Shield's built-in email-based activation flow within the `Auth` config file.

```php
public array $actions = [
    'login'    => null,
    'register' => \CodeIgniter\Shield\Authentication\Actions\EmailActivator,
];
```

> 

Author: The line breaks serve as a way to find the sections when scanning through.

    'users.create'        => 'Can create new non-admin users',
    'users.edit'          => 'Can edit existing non-admin users',
    'users.delete'        => 'Can delete existing non-admin users',
    'beta.access'         => 'Can access beta-level features',
];
```

### Assign Permissions to a Group

Each group can have its own specific set of permissions. These are defined in `Config\AuthGroups::$matrix`. You can specify each permission by it's full name, or using the context and an asterisk (*) to specify all permissions within that context.

```php
public array $matrix = [
    'superadmin' => [
        'admin.*',
        'users.*',
        'beta.access',
    ],
    //
];
```

### Assign Permissions to a User

Permissions can also be assigned directly to a user, irregardless of what groups they are a part of. This is done programatically on the `User` Entity.

```php
$user = auth()->user();

$user->addPermission('users.create', 'beta.access');
```

This will add all new permissions. You can also sync permissions so that the user ONLY has the given permissions directly assigned to them. Any not in the provided list are removed from the user.

```php
$user = auth()->user();

> $user->syncPermissions(['users.create', 'beta.access']);

Author: Yeah, I saw that. Luckily, we're still in beta. It's on my mental fix-list. :)

```

## Check If a User Has Permission

When you need to check if a user has a specific permission, you want to check both within the groups they are a part of, as well as any specifically assigned to them, you should use the `can()` method on the `User` entity.

```php
if (! auth()->user()->can('users.create')) {
    return redirect()->back()->with('error', 'You do not have permissions to access that page.');
}
```

This can also be done through a [controller filter](https://codeigniter.com/user_guide/incoming/filters.html) if you want to apply it to multiple pages of your site.

> 

Author: These serve as a visual indicator of section breaks when editing the file and scanning the page. It shouldn't cause any extra HTML, I don't believe.

## Managing Users

Since Shield uses a more complex user setup than many other systems, due to the [User Identities](1-concepts.md#identities), this quick overview should help you feel more confident when working with users on a day-to-day basis.

### Creating Users

By default, the only values stored in a user is the username. The first step is to create the user record with the username. If you don't have a username, be sure to set the value to `null` anyway, so that it passes CodeIgniter's empty data check.

```php
$users = model('UserModel');
$user = new User([
    'username' => 'foo-bar'
]);
$users->save($user);

// Get the updated user so we have the ID...
$user = $users->findById($users->getInsertID());

// Store the email/password identity for this user.
$user->createEmailIdentity($this->request->getPost(['email', 'password']));

// Add to default group
$users->addToDefaultGroup($user);
```

### Deleting Users

> A user's data can be spread over a few different tables so you might be concerned about how to delete all of the user's data from the system. This is handled automatically at the database level for all information that Shield knows about, through the `onCascade` settings of the table's foreign keys. You can delete a user like any other entity.

Reviewer: This is a bit misleading since our UserModel implements soft deletes by default. Please either update the example to be a hard delete or differentiate the process more.

```php
$users = model('UserModel');
$users->delete($user->id);
```

### Editing A User

> The `UserModel::save()` method has been modified to ensure that an email or password previously set on the `User` entity will be automatically in the correct `UserIdentity` record.

Reviewer: ```suggestion
The `UserModel::save()` method has been modified to ensure that an email or password previously set on the `User` entity will be automatically updated in the correct `UserIdentity` record.
```

```php
$users = model('UserModel');

> $user = $users->findById(123);

Reviewer: ```suggestion
$user  = $users->findById(123);
```

$user->fill([
    'username' => 'JoeSmith111',
    'email' => 'joe.smith@example.com',
    'password' => 'secret123'
]);
$users->save($user);
```

> If you prefer to user the `update()` method, than you will have to update the user's appropriate UserIdentity manually.

Reviewer: ```suggestion
If you prefer to use the `update()` method then you will have to update the user's appropriate UserIdentity manually.
```

```php
$users = model('UserModel');
$user = $users->findById(123);

$user->fill([
    'username' => 'JoeSmith111',
    'email' => 'joe.smith@example.com',
    'password' => 'secret123'
]);

// Saves the username field

> $users->save($user);

Reviewer: I think you meant for this to be an example of using `update()`, no?
```suggestion
$users->update($user);
```
